xls/modules/zstd/BUILD

```diff
@@ -213,8 +213,6 @@ cc_test(
     tags = ["manual"],
```

Comment:
Is there a reason why this test is marked `"manual"` ? Given that this was hiding an issue, it is probably good to have that enabled.
(Only if it is taking too many resources I'd consider it tagging manual. But then with a comment as explanation)

Reply:
We marked all DSLX and C++ tests as `manual` so that those would be called only in the separate [ZSTD-specific CI workflow](https://github.com/google/xls/blob/main/.github/workflows/modules-zstd.yml). It explicitly calls all of those rules tagged as `manual` and the workflow is triggered when there is a PR that changes the `xls/modules/zstd` directory or there is a push to `main` branch that changes it.

This was first requested in https://github.com/google/xls/pull/1315#issuecomment-2117654785.
After that we also decided to tag some of the `build` rules as `manual` because e.g. `place_and_route` rule is time consuming and it makes sense to not run it in every CI run (see https://github.com/google/xls/pull/1315#discussion_r1661866743).

Reply:
@proppy do the reasons for manually running tests still hold ? I think it is more a danger to not run tests all the time, because then hidden issues will go undetected.

Reply:
@hzeller the initial reason was that there were wip tests capturing current limitations of the implementation, but I don't believe that's the case anymore is it @lpawelcz ?

Reply:
@proppy Note you can disable them in gtest by calling them `TEST(MyTest, DISABLED_DoStuffThatOnlyCompilesButFailAtRuntime) { ... }`

Reply:
https://google.github.io/googletest/advanced.html#temporarily-disabling-tests

Reply:
> @hzeller the initial reason was that there were wip tests capturing current limitations of the implementation, but I don't believe that's the case anymore is it @lpawelcz ?

Yes, that was the initial reason that no longer holds.
There is also a second reason which is the length of the CI workflow. We wanted to avoid running all of our tests each time because both the C++ tests and the benchmarks (build rules for synthesis, place and route; run rules for IR benchmarks) are very time consuming.

With the suggestion from @cdleary I removed the `manual` tags for the C++ tests so that at least those will be executed in the main CI workflow.